### PR TITLE
Add submode

### PR DESCRIPTION
**Pull request by Werzi2001**
This is my first take on adding a submode as described in #68 . I checked and it works according to the ADIF standard (by adding <submode:X>) and in the cloudlog api (stations branch).

It is based on the #67 branch, not develop. If #67 gets merged I can adapt that.

What do you think about the changes?

### CODE REVIEW

**Comment by myzinsky** [comment]
Looks good and reasonable for me. Please adapt (if required) such that I can merge it.

**Comment by myzinsky** [comment]
Will that also still work if FLRIG sends USB or LSB?

**Comment by Werzi2001** [comment]
@myzinsky That's actually a good question. I don't have a working FLRIG setup. Do you know if/how I could test that?

**Comment by Werzi2001** [comment]
@myzinsky I updated the code to be "mergable" (not contains the other changes anymore that were already merged). I also checked the rig integration and adapted the `if(mode == "USB" || mode == "LSB") {` ins QSOView.qml. I don't know how to test it but I guess that is the part that is relevant for FLRIG.

**Comment by Werzi2001** [comment]
@myzinsky Any update on your side? Also concerning the release of the new version?

**Comment by myzinsky** [comment]
Have to test it with FLRIG, have some reports that it is broken...

File: src/csvtools.cpp [hunk 1]
```diff
@@ -20,13 +20,25 @@ QString csvTools::assemble(QString call,
...
+    }
+
     return QString("") +
            date + ", " +
            time + ", " +
            call + ", " +
            name + ", " +
            freq + ", " +
-           mode + ", " +
+           modeWithoutSubmode + ", " +
```
Comment (myzinsky) [inline src/csvtools.cpp:40]: Can you please use the name mode her? I think mode and submode would be enough :-), modeWithoutSubmode is a bit long IMHO

**Review by myzinsky — COMMENTED** [review]
Just a bit refactoring for the names...

File: src/csvtools.cpp (hunk 1 above)
Comment (Werzi2001) [reply src/csvtools.cpp:40]: @myzinsky Should I still change that? Just saw that you merged it anyway. Personally I prefer clear (even if long) names. As `mode` in the rest of the code means "mode + submode" I chose `modeWithoutSubmode` here.

Comment (myzinsky) [reply src/csvtools.cpp:40]: I will refactor it maybe after some time. 

Are you in Matrix? then you could join our discussion Chat there ...

Comment (Werzi2001) [reply src/csvtools.cpp:40]: @myzinsky Yes I have the chat.darc.de account and just joined the discussion there

**Pull request merged**